### DISCUSSION TITLE
Multi-GPU DDP training - performance loss

### DISCUSSION THREAD

**Discussion by oiao** [pyg-team/pytorch_geometric#9435] [Q&A]
Hi,

I have a model run graph-level predictions on a dataset with of ~1e4 small graphs, typically below 100 nodes each. 

Recently, I have started using it through Pytorch-Lightning's API, which also allowed for effortless training on multiple GPUs through the Distributed Data Parallel backbone.

Unfortunately, I noticed that models trained across multiple GPUs suffer from a significantly lower performance compared to a model that is only trained on one Unit. I was wondering if anyone else had a similar experience and found a way to mitigate this.
For reference, I am using the [`pyg.LightningDataset`](https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.data.lightning.LightningDataset.html) as a wrapper for my training/validation/test data, but from looking at the source code it does not seem like this matters too much.

Any hints would be appreciated!

**Comment by akihironitta** [comment]
Would you mind sharing your code and env details to reproduce the behaviour?
